presto-main/src/main/java/com/facebook/presto/operator/HttpTaskResultClient.java

```diff
@@ -69,467 +51,65 @@
-                                    log.debug("Unexpected acknowledge response code: %s", response.getStatusCode());
-                                }
-                                return null;
-                            }
-                        });
-                    }
-                }
-                catch (PrestoException e) {
-                    handleFailure(e, resultFuture);
-                    return;
-                }
-
-                // add pages:
-                // addPages must be called regardless of whether pages is an empty list because
-                // clientCallback can keep stats of requests and responses. For example, it may
-                // keep track of how often a client returns empty response and adjust request
-                // frequency or buffer size.
-                if (clientCallback.addPages(PageBufferClient.this, pages)) {
-                    pagesReceived.addAndGet(pages.size());
-                    rowsReceived.addAndGet(pages.stream().mapToLong(SerializedPage::getPositionCount).sum());
-                }
-                else {
-                    pagesRejected.addAndGet(pages.size());
-                    rowsRejected.addAndGet(pages.stream().mapToLong(SerializedPage::getPositionCount).sum());
-                }
-
-                synchronized (PageBufferClient.this) {
-                    // client is complete, acknowledge it by sending it a delete in the next request
-                    if (result.isClientComplete()) {
-                        completed = true;
-                    }
-                    if (future == resultFuture) {
-                        future = null;
-                    }
-                    lastUpdate = DateTime.now();
-                }
-                requestsCompleted.incrementAndGet();
-                clientCallback.requestComplete(PageBufferClient.this);
-            }
-
-            @Override
-            public void onFailure(Throwable t)
-            {
-                log.debug("Request to %s failed %s", uri, t);
-                checkNotHoldsLock(this);
-
-                t = rewriteException(t);
-                if (!(t instanceof PrestoException) && backoff.failure()) {
-                    String message = format("%s (%s - %s failures, failure duration %s, total failed request time %s)",
-                            WORKER_NODE_ERROR,
-                            uri,
-                            backoff.getFailureCount(),
-                            backoff.getFailureDuration().convertTo(SECONDS),
-                            backoff.getFailureRequestTimeTotal().convertTo(SECONDS));
-                    t = new PageTransportTimeoutException(fromUri(uri), message, t);
-                }
-                handleFailure(t, resultFuture);
-            }
-        }, pageBufferClientCallbackExecutor);
     }
 
-    private synchronized void sendDelete()
+    @Override
+    public void acknowledgeResults(long nextToken)
```

Comment:
I validated this implementation is refactored from 

https://github.com/prestodb/presto/blob/479e6552eff9f6e1776e3f40913d053e934241c5/presto-main/src/main/java/com/facebook/presto/operator/HttpPageBufferClient.java#L339-L357